### ISSUE TITLE
UndefinedBehaviorSanitizer: undefined-behavior /home/ldorau/work/rpma/src/mr.c:

### ISSUE THREAD

**Issue by ldorau** [pmem/rpma#303]
```
/home/ldorau/work/rpma/src/mr.c:264:2: runtime error: store to misaligned address 0x7ffe81e477e1 for type 'uint64_t' (aka 'unsigned long'), which requires 8 byte alignment
0x7ffe81e477e1: note: pointer points here
 7f 00 00  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  ff ff ff ff ff
              ^ 
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/ldorau/work/rpma/src/mr.c:264:2 in 
/home/ldorau/work/rpma/src/mr.c:266:2: runtime error: store to misaligned address 0x7ffe81e477e9 for type 'uint64_t' (aka 'unsigned long'), which requires 8 byte alignment
0x7ffe81e477e9: note: pointer points here
 03 02 01  00 ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  ff ff ff ff ff
              ^ 
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/ldorau/work/rpma/src/mr.c:266:2 in 
/home/ldorau/work/rpma/src/mr.c:268:2: runtime error: store to misaligned address 0x7ffe81e477f1 for type 'uint32_t' (aka 'unsigned int'), which requires 4 byte alignment
0x7ffe81e477f1: note: pointer points here
 0b 0a 09  08 ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  ff ff 84 4d dc
              ^ 
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/ldorau/work/rpma/src/mr.c:268:2 in 
/home/ldorau/work/rpma/src/mr.c:288:20: runtime error: load of misaligned address 0x7ffe81e477e1 for type 'uint64_t' (aka 'unsigned long'), which requires 8 byte alignment
0x7ffe81e477e1: note: pointer points here
 7f 00 00  ff 07 06 05 04 03 02 01  00 0f 0e 0d 0c 0b 0a 09  08 13 12 11 10 01 ff ff  ff ff ff ff ff
              ^ 
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/ldorau/work/rpma/src/mr.c:288:20 in 
/home/ldorau/work/rpma/src/mr.c:290:18: runtime error: load of misaligned address 0x7ffe81e477e9 for type 'uint64_t' (aka 'unsigned long'), which requires 8 byte alignment
0x7ffe81e477e9: note: pointer points here
 03 02 01  00 0f 0e 0d 0c 0b 0a 09  08 13 12 11 10 01 ff ff  ff ff ff ff ff ff ff ff  ff ff ff ff ff
              ^ 
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/ldorau/work/rpma/src/mr.c:290:18 in 
/home/ldorau/work/rpma/src/mr.c:292:18: runtime error: load of misaligned address 0x7ffe81e477f1 for type 'uint32_t' (aka 'unsigned int'), which requires 4 byte alignment
0x7ffe81e477f1: note: pointer points here
 0b 0a 09  08 13 12 11 10 01 ff ff  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  ff ff 84 4d dc
              ^ 
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /home/ldorau/work/rpma/src/mr.c:292:18 in 
```

**Comment by ldorau** [comment]
Reproduction:
The build:
https://travis-ci.org/github/ldorau/rpma/builds/722019251
and its jobs:
https://travis-ci.org/github/ldorau/rpma/jobs/722019252
https://travis-ci.org/github/ldorau/rpma/jobs/722019254
https://travis-ci.org/github/ldorau/rpma/jobs/722019253

**Comment by ldorau** [comment]
Reproduction: https://github.com/pmem/rpma/pull/309

**Comment by ldorau** [comment]
Fixed by #386
